osu.Desktop/OsuGameDesktop.cs

```diff
@@ -146,9 +159,60 @@ protected override string LocateBasePath()
                 return null;
             }
 
-            public StableStorage(DesktopGameHost host)
-                : base(string.Empty, host)
+            /// <summary>
+            /// osu!stable <c>Songs</c> folder can be moved to another location by changing the <c>BeatmapDirectory</c> setting in the stable configuration file.
+            /// This locates the <c>Songs</c> folder location for this stable installation and sets flags if the song folder location has been customized.
+            /// </summary>
+            private string locateSongsDirectory()
             {
+                //we get the user config file.
+                var configFile = GetStream(GetFiles(".", "osu!.*.cfg").First());
+                var textReader = new StreamReader(configFile);
+
+                while (!textReader.EndOfStream)
+                {
+                    var line = textReader.ReadLine();
+
+                    if (line?.StartsWith("BeatmapDirectory") == true)
+                    {
+                        var directory = line.Split('=')[1].TrimStart();
+
+                        if (!directory.Equals(default_songs_path, StringComparison.Ordinal))
+                        {
+                            usesCustomSongPath = true;
+                            return Path.GetFullPath(directory);
+                        }
+
+                        break;
+                    }
+                }
+
+                return default_songs_path;
+            }
+
+            public override Storage GetStorageForDirectory(string directory)
+            {
+                if (directory.Equals(default_songs_path, StringComparison.Ordinal) && usesCustomSongPath)
```

Comment:
```suggestion
                if (usesCustomSongPath && directory.Equals(default_songs_path, StringComparison.Ordinal))
```

put bool condition first